libsupport/include/katana/EntityTypeManager.h

```diff
@@ -359,6 +359,16 @@ class KATANA_EXPORT EntityTypeManager {
     return std::nullopt;
   }
 
+  /// \returns a vector containing all atomic type names
+  std::vector<std::string> GetAtomicTypeNames() const {
```

Comment:
This looks fine, but bear in mind this is only atomic types.  A node might have type` {Message, Post}`, which (confusingly) is the intersection of `Message `and `Post`.

Reply:
That's totally fine from my point of view. I'm adding this because types ultimately correspond to different feature representations for the ML stuff. How to handle intersection types for feature purposes can/will probably be handled in python eventually.

Reply:
BTW, I would not right an intersection type like that because it is confusing. I would write `Message & Post`.

Reply:
I think you are right that `Message & Post` is a better model, though it still doesn't scream "intersection" at me.  I wrote `{Message, Post}` because types get passed to graph update as sets of strings.

Reply:
re: `&` as intersection: If I write `a & b` (where `a` and `b` are say 64-bit integers) the set of 1-bits in the result will be exactly the intersection of the 1-bits in `a` and `b`. Bitwise and *is* intersection of small finite sets. Just as bitwise or is union.

I'm also happy to use `Message ∩ Post` if you prefer though. Unicode FTW.

Reply:
Fan boy rant: Scala 3 supports type intersections using `&`. https://docs.scala-lang.org/scala3/book/types-intersection.html It's quite lovely.

Reply:
I thought @aneeshdurg had a PR that added this API and I instead wanted an iterator over `std::string` that is build out of the iterators over the `map`.  See `boost::transform_iterator`. 

@arthurp : I see that folks are asking to iterate over `string` names, and I think an iterator is a good compromise, where we don't stick to building a vector but we also provide a cheap method of iteration over `strings`. By forcing folks to use `EntityTypeIDs` , we may be forcing duplication of not so straightforward code that would have been straightforward given the right API.

Reply:
@astolman : please check with @aneeshdurg as he wrote the exact same code in a PR the other day (if I recall).

Reply:
This seems orthogonal to me. Iterators vs vectors and names vs IDs are separate issues. I'm not worried about performance really, I'm worried about separation of concern. I don't think there will be any duplicated code. We can have a "list converter" similar to `GetEntityTypeIDs` (see below) which converts a contain of one type to another, to work around the lack of an idiomatic way to do iterator comprehensions in C++.

https://github.com/KatanaGraph/katana/blob/e8cfb244812db5bd561b2e8c4af37162433b4c21/libsupport/include/katana/EntityTypeManager.h#L291-L308

Side note: I would prefer a range to an interator.

I'm clearly in the minority in this debate though, so if strings are where people want it, so be it. I do reserve the right to laugh when weird bugs crop up or people have to start converting IDs to strings to call methods, though. ;-)

Reply:
I agree that the two issues are orthagonal. Secondly, by `iterators` I mean `iterator ranges` for now, until we arrive at C++-20's ranges. 

TBH, I haven't spent time looking carefully at the various use cases of `EntityTypeManager's API`. But I have heard @witchel asking for the same, so I took the position of "the interface should provide it in an efficient yet generic manner". It could be that these folks are wrong and you are right and you will be laughing at us :).